### PR TITLE
MULE-9967: config-ref accept MEL expression and accept invalid values

### PR DESCRIPTION
**Pull request by marianogonzalez**

### CODE REVIEW

File: modules/extensions-spring-support/src/test/resources/operation-with-expression-config-ref.xml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+                <heisenberg:previous>
+                    <heisenberg:door victim="Krazy-8" address="Jesse's"/>
+                </heisenberg:previous>
+            </heisenberg:door>
+        </heisenberg:next-door>
+
+    </heisenberg:config>
+
+    <flow name="sayMyName">
+        <heisenberg:say-my-name config-ref="#[dynamic]"/>
```
Comment (elrodro83) [inline modules/extensions-spring-support/src/test/resources/operation-with-expression-config-ref.xml]: is this name 'dynamic' supposed to match something on this xml?

File: modules/extensions-spring-support/src/test/resources/source-with-expression-config-ref.xml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+                       recipe="recipes"
+                       initialHealth="CANCER"
+                       finalHealth="CANCER"
+                       monthlyIncomes="monthlyIncomes"
+                       candidateDoors="candidateDoors"
+                       enemies="#[[]]">
+    </heisenberg:config>
+
+    <flow name="source" initialState="stopped">
+        <heisenberg:listen-payments initialBatchNumber="0" config-ref="#[dynamic]" />
```
Comment (elrodro83) [inline modules/extensions-spring-support/src/test/resources/source-with-expression-config-ref.xml]: is this name 'dynamic' supposed to match something on this xml?

File: core/src/main/java/org/mule/runtime/core/util/TemplateParser.java [hunk 3]
```diff
@@ -51,11 +51,6 @@
      */
     protected static final Logger logger = LoggerFactory.getLogger(TemplateParser.class);
 
-    public static final Pattern ANT_TEMPLATE_PATTERN = patterns.get(ANT_TEMPLATE_STYLE).getPattern();
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/util/TemplateParser.java]: these are no longer used, right?

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java [hunk 4]
```diff
@@ -53,7 +60,9 @@
 public abstract class ExtensionComponent implements MuleContextAware, MetadataAware, FlowConstructAware, Initialisable
 {
 
+    private final TemplateParser expressionParser = TemplateParser.createMuleStyleParser();
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java]: static import

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java [hunk 5]
```diff
@@ -42,6 +47,8 @@
 
 import javax.inject.Inject;
 
+import org.apache.logging.log4j.core.config.ConfigurationException;
```
Comment (elrodro83) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java]: ?

File: core/src/main/java/org/mule/runtime/core/util/TemplateParser.java (hunk 3 above)
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/util/TemplateParser.java]: right

File: modules/extensions-spring-support/src/test/resources/operation-with-expression-config-ref.xml (hunk 1 above)
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/resources/operation-with-expression-config-ref.xml]: nop. just a random expression to make sure the validation is triggered

File: modules/extensions-spring-support/src/test/resources/source-with-expression-config-ref.xml (hunk 2 above)
Comment (marianogonzalez) [inline modules/extensions-spring-support/src/test/resources/source-with-expression-config-ref.xml]: idem

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java (hunk 5 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java]: damn... fixed

File: modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java (hunk 4 above)
Comment (marianogonzalez) [inline modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/ExtensionComponent.java]: fixed

**Pull request merged**